Automattic/wp-calypso
# Launchpad: Clean up task-helper

Related to #86647

## Proposed Changes
* Remove the ~700 lines from the task-helper `getEnhancedTasks` 
* Move the getEnhancedTasks to a separate file
* Migrate the missing dependencies to the context they are used.
* Transform siteInfoQueryArgs to a function

### Review instructions
I recommend using the commits to follow the changes. 

* I started moving the big switch from the task helper b2093cc0f6bb5b6d57cecf1cebb9d216c59a04de
* I moved to the correct context  https://github.com/Automattic/wp-calypso/pull/86960/commits/aa0498e61d490c2ebc9612a6f33630237427363e#diff-1af91985184f6d7b895325749a29ebd09bd8f1e4a570a3fa86cc9fb892416c3fL90-L92 
* I moved the siteInfoQueryArgs to a function so it can be shareable between tasks df60d3cc144d1bc99c950594bd20c3e8c848c9f2
* So I extract the getEnhancedTasks from task helper to a separate file. df60d3cc144d1bc99c950594bd20c3e8c848c9f2

### Testing instructions
Since all flows are already reviewed with the new code we can just run a smoke test in 1-2 of them to check if the new code is working, completing the flow.

* You can use the `setup/free` . More instructions are available https://github.com/Automattic/wp-calypso/pull/86432

## Pre-merge Checklist
